flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperator.java

```diff
@@ -551,19 +657,28 @@ public void registerProcessingTimeTimer(long time) {
 			}
 		}
 
+		void registerCleanupTimer(long time) {
```

Comment:
The logic for these two (`register/deleteCleanupTimer`) should be moved directly into `WindowOperator`.